checks.d/yarn.py

```
@@ -0,0 +1,264 @@
+    'finishedTime': ('yarn.apps.finishedTime', GAUGE),
+    'elapsedTime': ('yarn.apps.elapsedTime', GAUGE),
+    'allocatedMB': ('yarn.apps.allocatedMB', GAUGE),
+    'allocatedVCores': ('yarn.apps.allocatedVCores', GAUGE),
+    'runningContainers': ('yarn.apps.runningContainers', GAUGE),
+    'memorySeconds': ('yarn.apps.memorySeconds', GAUGE),
+    'vcoreSeconds': ('yarn.apps.vcoreSeconds', GAUGE),
+}
+
+YARN_NODE_METRICS = {
+    'lastHealthUpdate': ('yarn.node.lastHealthUpdate', GAUGE),
+    'usedMemoryMB': ('yarn.node.usedMemoryMB', GAUGE),
+    'availMemoryMB': ('yarn.node.availMemoryMB', GAUGE),
+    'usedVirtualCores': ('yarn.node.usedVirtualCores', GAUGE),
+    'availableVirtualCores': ('yarn.node.availableVirtualCores', GAUGE),
+    'numContainers': ('yarn.node.numContainers', GAUGE),
+}
+
+
+class YarnCheck(AgentCheck):
+
+    def check(self, instance):
+
+        # Build the resourceManager URI
+        resourcemanager_hostname = instance.get('resourcemanager_address', DEFAULT_HOSTNAME)
+        resourcemanager_port = instance.get('resourcemanager_port', DEFAULT_RESOURCEMANAGER_PORT)
+        rm_address = resourcemanager_hostname + ':' + str(resourcemanager_port)
+
+        # Get the cluster ID from Yarn
+        cluster_id = self._get_cluster_id(rm_address)
+
+        # If a Cluster ID was not found, assign it 0
+        if not cluster_id:
+            cluster_id = 0
+
+        self._yarn_cluster_metrics(resourcemanager_hostname, cluster_id, rm_address)
+        self._yarn_app_metrics(resourcemanager_hostname, rm_address)
+        self._yarn_node_metrics(resourcemanager_hostname, cluster_id, rm_address)
+
+    def _get_cluster_id(self, rm_address):
+
+        # Get the cluster info
+        info_json = self._rest_request_to_json(rm_address, INFO_URI)
+
+        if info_json:
+            if info_json['clusterInfo']:
+                if info_json['clusterInfo']['id']:
+                    return info_json['clusterInfo']['id']
+
```

Should we raise an exception here ?
```python
def _get_cluster_id(self, rm_address):
    info_json = self._rest_request_to_json(rm_address, INFO_URI)
    cluster_id = info_json.get("clusterInfo", {}).get("id")
    if cluster_id is not None:
        return cluster_id
    raise Exception("Unable to retrieve cluster id")
```